packages/@lightningjs/ui-components/src/globals/context/theme-manager.js

```diff
@@ -37,7 +37,79 @@ const merge = {
     return result;
   }
 };
+
+const base64Cache = [];
+
 const isSubTheme = themeName => 'subTheme' === themeName.slice(0, 8);
+
+/**
+ * Extracts the MIME type from a Data URI.
+ *
+ * @param {string} dataUri - The Data URI string.
+ * @returns {string|null} The extracted MIME type, or null if not found.
+ */
+function getMimeTypeFromDataUri(dataUri) {
+  const matches = dataUri.match(/^data:(.*?);base64,/);
+  if (matches && matches.length === 2) {
+    return matches[1];
+  }
+  return null;
+}
+
+/**
+ * Checks if a string represents a Base64-encoded image and extracts the MIME type.
+ *
+ * @param {string} str - The string to check.
+ * @returns {{ isImage: boolean, mimeType: string|null }} An object indicating whether the string is an image and the extracted MIME type.
+ */
+function checkBase64EncodedImage(str) {
+  const regex = /^data:image\/(jpeg|jpg|png|gif);base64,/;
+  const isImage = regex.test(str);
+  const mimeType = isImage ? getMimeTypeFromDataUri(str.match(regex)[0]) : null;
+
+  return {
+    isImage,
+    mimeType
+  };
+}
+
+/**
+ * Converts a Base64-encoded image to a Blob URL.
+ * Note: Make sure to handle potential memory leaks caused by the browser's image caching.
+ *
+ * @param {string} base64String - The Base64-encoded image string.
+ * @param {string} mimeType - The MIME type of the image.
+ * @returns {string|null} The Blob URL representing the converted image, or null if conversion fails.
+ */
+function base64ToBlobURL(base64String, mimeType) {
+  const byteCharacters = atob(
+    base64String.substring(base64String.indexOf(',') + 1)
+  );
+  const byteArrays = [];
+
+  try {
+    for (let offset = 0; offset < byteCharacters.length; offset += 512) {
+      const slice = byteCharacters.slice(offset, offset + 512);
+      const byteNumbers = new Array(slice.length);
+
+      for (let i = 0; i < slice.length; i++) {
+        byteNumbers[i] = slice.charCodeAt(i);
+      }
+
+      const byteArray = new Uint8Array(byteNumbers);
+      byteArrays.push(byteArray);
+    }
+
+    const blob = new Blob(byteArrays, { type: mimeType });
+    const blobURL = URL.createObjectURL(blob);
+
+    return blobURL;
+  } catch (error) {
+    logger.info('Unable to convert base64 image to URL');
```

Comment:
do we want to include the stack trace here?

Reply:
Good point, we probably should. Would it be go to add a follow up ticket to add stack trace support to all of our logs?